### PR TITLE
add flow-edges from users with trusted tokens to organizations

### PR DESCRIPTION
**Pull request by jaensen**

### CODE REVIEW

File: src/safe_db/db.rs [hunk 1]
```diff
@@ -31,6 +31,45 @@ impl DB {
...
+            }
+        }
+
+        // Find all safes that have a non-zero balance of tokens that are accepted by an organization
+        for (user, safe) in &self.safes {
+            for (token, balance) in &safe.balances {
+                if balance == &U256::from(0) {
+                    continue;
+                }
+                organization_accepted_tokens.get(token).map(|organizations| {
```
Comment (chriseth) [inline src/safe_db/db.rs]: Does this compile? The function passed to `map` does not return anything.
Maybe `if let Some(organizations) = rganization_accepted_tokens.get(token) {`?

Comment (chriseth) [reply src/safe_db/db.rs]: or use something like `flat_map` and `for_each`

**Comment by chriseth** [comment]
Does it work? :)

File: src/safe_db/db.rs (hunk 1 above)
Comment (jaensen) [reply src/safe_db/db.rs]: `organization_accepted_tokens.get(token)` returns an `Option<HashSet<Address>>` so the method used is https://doc.rust-lang.org/std/option/enum.Option.html#method.map and not https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.map

File: src/safe_db/db.rs [hunk 2]
```diff
@@ -31,6 +31,45 @@ impl DB {
...
+                if receiver_safe.organization {
+                    //println!("user {} can send {} token to orga {}", user, safe.token_address, send_to);
+                    organization_accepted_tokens.entry(safe.token_address).or_default().insert(*send_to);
+                }
+            }
+        }
+
+        // Find all safes that have a non-zero balance of tokens that are accepted by an organization
+        for (user, safe) in &self.safes {
+            for (token, balance) in &safe.balances {
```
Comment (llunaCreixent) [inline src/safe_db/db.rs:58]: **Meaning of `balance.token`**
It's not clear to me what the `token` attribute of an `Edge` represents. Is it a token address, or a user address (i.e. the owner of the token)? Compare these 2 cases, where in the first it is assumed that the balance contains the token address, and in the second the `token` attribute of the balance is supposed to contain the token owner address:
- https://github.com/CirclesUBI/pathfinder2/blob/5950ab81b2d1adf36354e3b9e7bc5b16e0a97874/src/safe_db/db.rs#L94C1-L103C1
- https://github.com/CirclesUBI/pathfinder2/blob/5950ab81b2d1adf36354e3b9e7bc5b16e0a97874/src/safe_db/db.rs#L55C1-L68C4

I think the problem lies in the confusion created by the `edge.token` which we are using to store the token owner address. Correct me if I'm wrong. I suggest changing the name of the `edge.token` attribute to `edge.token_owner` once and for all.

Comment (JacqueGM) [reply src/safe_db/db.rs:58]: can we create a test image to update the pathfinder in production  @jaensen ?

Comment (llunaCreixent) [reply src/safe_db/db.rs:58]: In my PR (https://github.com/CirclesUBI/pathfinder2/pull/43), I assumed that `balance.token` is a token address and not a token's owner address. So the issue reported in my comment would be fixed there.

**Comment by JacqueGM** [comment]
already added in the only-server

**Pull request closed without merging**